### PR TITLE
feat: Update odtp.yml to v0.5.0

### PR DESCRIPTION
**Pull request by caviri**
Update odtp.yml to newest odtp.yml version. This includes a better description of authors and tools used, information about dockerhub hosted images, and a proposal on dynamic naming convention:

v0.5.0 `odtp.yml` proposal is here
https://github.com/odtp-org/odtp-component-template/blob/main/odtp.yml

### CODE REVIEW

File: odtp.yml [hunk 1]
```diff
@@ -67,4 +85,5 @@ schema-output: null
 
 # If applicable, define devices needed such as GPU.
 devices:
-  gpu: false
\ No newline at end of file
+  - type: gpu
+    required: true
```
Comment (sabinem) [inline odtp.yml]: **nitpick**: Please add a blank line at the end of the file

Comment (sabinem) [inline odtp.yml]: **fix**: should this be `false`? as this component does not require a GPU?

**Review by sabinem — CHANGES_REQUESTED** [review]
**suggestion**: generally I would prefer to have a flatter structure, especially for the operational important variables such as `outputs` and `devices`. 

**thought**: I need the structure of `odtp.yml` to be stable and the same for all components as we are currently making it operational. So in case you are sure we need that change now, it is okay and I will adapt my parsing, but I would like to have it stable during my ODTP development phase in December: so we can integrate this now in case you would update the other components as well. Or we update this in January after I finished my development on the `odtp.yml` integration.

**todo**: fix the `gpu: true` as this component does not need a gpu. After that I will approve it.

**Comment by caviri** [comment]
Hi @sabinem, I'm not so fully convinced of the `outputs`/`inputs` section. I was hoping to start the discussion via this repository. The main issue is how to represent outputs files names that are generated programatically via some parameters outputs. I know Robin was working on this from the semantic side but nothing was achieved so far. I'm open to suggestions. 

Nevertheless, I think that this version incorporated important sections such as a better authoring (list instead of a single line with all authors), dockerhub link, and an schema version which I think it's essential when dealing with the parsing. 

I think that the version will help you to organize the parsing of the `odtp.yml` as it can be use to validate the datamodel and open the door to compatibility with different minor versions. 

Would be OK for you if I roll back the sections regarding output/inputs but keep the others? I have a couple hours today that I can devote to this.

**Comment by sabinem** [comment]
@caviri Yes I would be fine with anything stable. I would just avoid going back and forth. If you are certain about the changes and have time to adapt the other components as well go for it. And if you are unsure about the `outputs` then just leave them as they are for now.

**Comment by caviri** [comment]
Hello @sabinem, I updated the odtp.yml and replied to your comments. Now it should be valid.

**Review by sabinem — APPROVED** [review]
**praise**: approved: you can merge this now

**Pull request merged**